pandas/core/groupby/groupby.py

```diff
@@ -2438,6 +2439,9 @@ def pre_processor(vals: ArrayLike) -> tuple[np.ndarray, np.dtype | None]:
             elif is_timedelta64_dtype(vals.dtype):
                 inference = np.dtype("timedelta64[ns]")
                 out = np.asarray(vals).astype(float)
+            elif isinstance(vals, ExtensionArray) and is_float_dtype(vals):
+                inference = np.dtype(np.float64)
```

Comment:
i think you need to make this `np.dtype(object)` and it will match

Reply:
nope, didn't work

Reply:
@jreback - hope this looks fine now